microsetta_admin/server.py

```diff
@@ -774,6 +775,191 @@ def scan():
                                       recorded_type)
 
 
+def get_legends(criteria):
+
+    dict = {}
+    hex = 'ABCDEF0123456789'
+    if criteria not in session:
+        status, fields = APIRequest.get(
+            '/api/admin/barcode_query_fields')
+
+        if status == 200:
+            for obj in fields:
+                if criteria == "Type" and obj["label"] == "Sample Site":
+                    for val in obj["values"]:
+                        code = [random.choice(hex) for i in range(6)]
+                        color_code = ['#' + ''.join(code)]
+                        while color_code[0] in dict:
+                            code = [random.choice(hex) for i in range(6)]
+                            color_code = ['#' + ''.join(code)]
+                        dict[obj["values"][val]] = color_code[0]
```

Comment:
Two issues here:
1) This code is essentially repeated twice below. Please decompose it into a function that only handles the task of generating a dictionary of colors.
2) The code can produce colors that are very similar, which somewhat defeats the purpose of color-coding. Can you consider alternatives that would do a better job of ensuring visual clarity?

Reply:
Created a function to generate random color code. The logic to generate color codes is adjusted so that produced colors are not very similar. The code is further simplified

Reply:
It's still generating very similar colors, see below.

![bulkscan_20230330](https://user-images.githubusercontent.com/83246693/228936891-a8b47a26-e127-4b71-86eb-17ef7d908f4d.png)